src/coreclr/jit/compiler.hpp

```diff
@@ -61,6 +61,14 @@ inline float forceCastToFloat(double d)
 // Enforce UInt32 narrowing for buggy compilers (notably Whidbey Beta 2 LKG)
 inline UINT32 forceCastToUInt32(double d)
 {
+#if defined(_MSC_VER) && defined(TARGET_X86)
+    // MSVC x86 emits __ftoui3 call for (UINT32)d
+    // and it returns 4294967295U for float.MaxValue instead of 0.
+    if (d == 3.40282346638528859e+38)
```

Comment:
Do we have tests covering these scenarios to ensure they are consistent across both current and future compilers?

Reply:
Is the same not needed for `float`?

Reply:
float->uint casts are forced through double on x86

https://github.com/dotnet/runtime/blob/fdc8148e9a8626da75f98f462f814946af35e5f1/src/coreclr/jit/morph.cpp#L175-L180

Reply:
Ah I see. I wonder if that's holdover from when we still used the x87 FPU stack